storage/ndb/src/kernel/vm/mt.cpp

```diff
@@ -3930,13 +3937,24 @@ thr_send_threads::assist_send_thread(bool must_send,
   }
   else if (!send_instance->m_awake)
   {
+    /**
+     * The send thread is not awake, so obviously not overloaded,
+     * we can continue our work, no need to assist.
+     */
     wakeup(&(send_instance->m_waiter_struct));
-    if (!must_send)
-    {
-      NdbMutex_Unlock(send_instance->send_thread_mutex);
-      return false;
-    }
+    NdbMutex_Unlock(send_instance->send_thread_mutex);
+    return false;
   }
+  else if (send_instance->m_num_trps_ready <= 3)
```

Comment:
[Just for noting]
This is an optimization to make assist sending not that eager,  but here still need to acquire send_thread_mutex which would cause lock contend. Maybe we can discuss it on workday.

Reply:
ok